rllib/algorithms/algorithm.py

```diff
@@ -616,7 +616,8 @@ def setup(self, config: AlgorithmConfig) -> None:
 
         # Evaluation WorkerSet setup.
         # User would like to setup a separate evaluation worker set.
-        if self.config.evaluation_num_workers > 0 or self.config.evaluation_interval:
+        # Note: We skipp workerset creation if we need to do offline evaluation
```

Comment:
typo: skip

Reply:
Dumb question: Is there any chance someone might want to do both? offline estimation on a dataset (one run through the complete dataset per iteration) AND have n eval rollout workers take some shot at an actual env? Just for research/debugging reasons.

What I'm saying is that we might want to consider separating the concepts of RolloutWorker/WorkerSet entirely from the concept of a separate dataset reader for offline data intake. Just some things to think about for the future.

Reply:
There could be a use-case like that. Right now we don't support it. If we want to support it in the future we should create a separate project and re-define the evaluation config for completely isolating the two (or even more) eval flows.

Reply:
Do we even support accepting both today? afaik the input config is like the following:
```
# for offline
{
    "input": "dataset",
    "input_config": {"format": xxx, "path": [path1, path2]}
}

# for online
{
    "input": "sampler",
    "input_config": {"env": "CartPole-v0"}
}
```

Reply:
Yeah, let's consolidate the entire input config API. I feel like there should be a single config method: `input_setup` or whatever name, which covers both our current `environment()` method as well as `rollouts()` (and `offline_data()` 😵‍💫 !).

To answer your question: Yeah, we have the mixed input, but it's not well explored/tested/configurable:
```
        >>> MixedInput({
        ...    "sampler": 0.4,
        ...    "/tmp/experiences/*.json": 0.4,
        ...    "s3://bucket/expert.json": 0.2,
        ... }, ioctx)
```

Reply:
yeah this will fall back to the old json reader I think which we want to get rid of anyway.